pkg/apis/settings/v1alpha1/defaults_cluster_openidconnect_preset_test.go

```diff
@@ -0,0 +1,100 @@
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package v1alpha1_test
+
+import (
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	"github.com/gardener/gardener/pkg/apis/settings/v1alpha1"
+)
+
+var _ = Describe("ClusterOpenIDConnectPreset defaulting", func() {
+
+	var (
+		given    *v1alpha1.ClusterOpenIDConnectPreset
+		expected *v1alpha1.ClusterOpenIDConnectPreset
+	)
+
+	BeforeEach(func() {
+
+		given = &v1alpha1.ClusterOpenIDConnectPreset{}
+		usernameClaim := "sub"
+		expected = &v1alpha1.ClusterOpenIDConnectPreset{
+			Spec: v1alpha1.ClusterOpenIDConnectPresetSpec{
+				OpenIDConnectPresetSpec: v1alpha1.OpenIDConnectPresetSpec{
+					Server: v1alpha1.KubeAPIServerOpenIDConnect{
+						// string literal are used to be sure that the test fails
+						// if the constant values are changed.
+						UsernameClaim: &usernameClaim,
+						SigningAlgs:   []string{"RS256"},
+					},
+					ShootSelector: &metav1.LabelSelector{},
+				},
+				ProjectSelector: &metav1.LabelSelector{},
+			},
+		}
+	})
+
+	It("should default ClusterOpenIDConnectPreset correctly", func() {
+		v1alpha1.SetDefaults_ClusterOpenIDConnectPreset(given)
+
+		Expect(given).To(BeEquivalentTo(expected))
+	})
+
+	It("should not default ProjectSelector if it is already set", func() {
+		given.Spec.ProjectSelector = &metav1.LabelSelector{MatchLabels: map[string]string{"foo": "bar"}}
+
+		v1alpha1.SetDefaults_ClusterOpenIDConnectPreset(given)
+
+		expected.Spec.ProjectSelector = &metav1.LabelSelector{MatchLabels: map[string]string{"foo": "bar"}}
+		Expect(given).To(BeEquivalentTo(expected))
+
+	})
+
+	It("should not default ShootSelector if it is already set", func() {
+		given.Spec.ShootSelector = &metav1.LabelSelector{MatchLabels: map[string]string{"foo": "bar"}}
+
+		v1alpha1.SetDefaults_ClusterOpenIDConnectPreset(given)
+
+		expected.Spec.ShootSelector = &metav1.LabelSelector{MatchLabels: map[string]string{"foo": "bar"}}
+		Expect(given).To(BeEquivalentTo(expected))
+
+	})
+
+	It("should not default SigningAlgs if they are already set", func() {
+		given.Spec.Server.SigningAlgs = []string{"alg1", "alg2"}
+
+		v1alpha1.SetDefaults_ClusterOpenIDConnectPreset(given)
+
+		expected.Spec.Server.SigningAlgs = []string{"alg1", "alg2"}
+		Expect(given).To(BeEquivalentTo(expected))
+
+	})
+
+	It("should not default UsernameClaim if it is already set", func() {
+		usernameClaim := "usr"
+		given.Spec.Server.UsernameClaim = &usernameClaim
```

Comment:
```suggestion
		given.Spec.Server.UsernameClaim = pointer.String("usr")
```

Similar changes at other places as well